pylint/checkers/classes/class_checker.py

```diff
@@ -880,6 +887,39 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
         self._check_proper_bases(node)
         self._check_typing_final(node)
         self._check_consistent_mro(node)
+        self._check_declare_non_slot(node)
+
+    def _check_declare_non_slot(self, node: nodes.ClassDef) -> None:
+        if not self._has_valid_slots(node):
+            return
+
+        slot_names = self._get_classdef_slots_names(node)
+
+        for base in node.bases:
+            ancestor = safe_infer(base)
+            if not isinstance(ancestor, nodes.ClassDef):
+                continue
+            if not self._has_valid_slots(ancestor):
+                return
+            else:
+                slot_names.extend(self._get_classdef_slots_names(ancestor))
```

Comment:
```suggestion
                for slot_name in self._get_classdef_slots_names(ancestor):
                    if slot_name == "__dict__":
                        return
                    slot_names.append(slot_name)
```
If I understood the code well, it seems we can cut all the following inferences if the a base contains ``__dict__`` which should be better performance wise as inference is costly. (based on intuition, not benchmark)

Reply:
Yes, this looks like a better idea to me, thanks!